code/modules/clothing/shoes/magboots.dm

```diff
@@ -57,3 +57,27 @@
 	name = "blood-red magboots"
 	icon_state = "syndiemag0"
 	magboot_state = "syndiemag"
+
+/obj/item/clothing/shoes/magboots/crushing
+	desc = "Normal looking magboots that are altered to increase magnetic pull to crush anything underfoot."
+
+/obj/item/clothing/shoes/magboots/crushing/proc/crush(mob/living/user)
+	if (!isturf(user.loc) || !magpulse)
+		return
+	var/turf/T = user.loc
+	for (var/mob/living/A in T)
+		if (A != user && A.lying)
+			A.adjustBruteLoss(rand(10,13))
+			to_chat(A,"<span class='userdanger'>[user]'s magboots press down on you, crushing you!</span>")
+			A.emote("scream")
+
+/obj/item/clothing/shoes/magboots/crushing/equipped(mob/user,slot)
+	. = ..()
+	if (slot == SLOT_SHOES)
```

Comment:
For efficiency, I think these signals should be registered/unregistered on attack_self() (magpulse activation/disactivation), instead of early returning crush() if magpulse is not TRUE.

So make sure to add a magpulse check here once that's is done.

Reply:
true

Reply:
actually no, because then if they turn it on and then unequip it it'll still call it, if I check if it's actually on them in crush() and unregister if not, it'll not work if they put it back on while it's still activated, so they'll have to turn it back on, and it'll only work for that specific person.